### PR TITLE
allow vault roleId to come from k8s Secret

### PR DESCRIPTION
**Pull request by intrand**
Implements https://github.com/external-secrets/external-secrets/issues/558

### CODE REVIEW

**Comment by gusfcarvalho** [comment]
Hi @intrand ! can you sign your commits showing that you accept our [Code of Conduct](https://external-secrets.io/v0.7.1/contributing/coc/)? :)

File: apis/externalsecrets/v1beta1/secretstore_vault_types.go [hunk 1]
```diff
@@ -124,7 +124,13 @@ type VaultAppRole struct {
 
 	// RoleID configured in the App Role authentication backend when setting
 	// up the authentication backend in Vault.
-	RoleID string `json:"roleId"`
+	RoleID string `json:"roleId,omitempty"`
```
Comment (gusfcarvalho) [inline apis/externalsecrets/v1beta1/secretstore_vault_types.go:127]: In order to make this optional to kubebuilder itself, you need to add a comment to flag this arg as optional:
```suggestion
        //+optional
	RoleID string `json:"roleId,omitempty"`
```

File: pkg/provider/vault/vault.go [hunk 2]
```diff
@@ -1241,7 +1241,21 @@ func revokeTokenIfValid(ctx context.Context, client Client) error {
...
+	if appRole.RoleID != "" { // use roleId from CRD, if configured
+		roleID = strings.TrimSpace(appRole.RoleID)
+	} else if appRole.RoleRef != emptySecretKeySelector { // use roleID from Secret, if configured
+		roleID, err = v.secretKeyRef(ctx, &appRole.RoleRef)
+		if err != nil {
+			return err
+		}
+	} else { // we ran out of ways to get roleId. return an appropriate error
+		return errors.New("one of RoleID or RoleRef are required")
+	}
```
Comment (gusfcarvalho) [inline pkg/provider/vault/vault.go:1258]: We also need to implement this check at the `ValidateStore` method, as we should run this check on the webhook level.

Comment (gusfcarvalho) [inline pkg/provider/vault/vault.go:1258]: Can you also add tests capturing this logic here?

Comment (intrand) [reply pkg/provider/vault/vault.go]: Thanks for pointing this out! I believe I've satisfied validation in `ValidateStore()` as of the latest commit 🤔 I'm still getting acquainted with the codebase, so thanks for being patient with me.

**Comment by gusfcarvalho** [comment]
Also, after implementing the tests, make sure to run `make reviewable` as it will make sure that all the files are up-to-date. To fix the linter issue, you can run `make fmt` command.

**Comment by intrand** [comment]
Thanks for the view thusfar @gusfcarvalho :) I'm going to stop for the day, but should hopefully find some time to pick this back up soon(TM).

**Comment by intrand** [comment]
I'm having a lot of trouble getting a test written for this. There's a lot going on inside the current function, and I don't see any other tests to base my tests off of. Would you, or someone else, be able to help?

**Comment by gusfcarvalho** [comment]
@intrand are you on kubernetes slack? if so, ping me @gusfcarvalho so we can schedule a call to go through it.

**Comment by gusfcarvalho** [comment]
hi @intrand ! any news? 😄

**Comment by intrand** [comment]
Hey there @gusfcarvalho sorry, but unfortunately not. I don't know how much more time I'll have to work on this in the next many months. If you'd like to close the PR or find someone to write some tests, please feel free. I have no intention of deleting the branch on my fork anytime soon.

**Comment by bdrich** [comment]
@gusfcarvalho @intrand 
Hoping this request will address testing request: #2284

**Comment by gusfcarvalho** [comment]
closing this in favor of #2284

**Pull request closed without merging**